pkg/deploy/registry/deploylog/rest.go

```diff
@@ -0,0 +1,206 @@
+package deploylog
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/rest"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/controller/framework"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	genericrest "github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic/rest"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/pod"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
+	"github.com/golang/glog"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+)
+
+// REST is an implementation of RESTStorage for the api server.
+type REST struct {
+	Deploy         DeploymentClient
+	PodGetter      pod.ResourceGetter
+	ConnectionInfo kclient.ConnectionInfoGetter
+	Timeout        time.Duration
+}
+
+// DeploymentClient defines a local interface to a deployment client for testability.
+type DeploymentClient interface {
+	GetDeploymentConfig(ctx kapi.Context, name string) (*deployapi.DeploymentConfig, error)
```

Comment:
Is a `DeploymentConfigNamespacer` enough? I mostly need to interact with the deployments (list, watch) so I guess I need a `ReplicationControllersNamespacer` as well, right?